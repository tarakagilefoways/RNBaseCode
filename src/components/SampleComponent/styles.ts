import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  input: {
    height: 40,
    width: '80%',
    alignSelf: 'center',
    fontSize: 18,
    borderWidth: 0.5,
    padding: 4,
    marginTop: 8,
  },
});
export default styles;
