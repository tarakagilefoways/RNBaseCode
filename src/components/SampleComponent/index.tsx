import React from 'react';
import { TextInput, TextInputProps, View } from 'react-native';
import styles from './styles';
interface Props extends TextInputProps {}
const InputComponent = (props: Props) => {
  const { ...restProps } = props;
  return (
    <View>
      <TextInput style={styles.input} {...restProps} />
    </View>
  );
};
export default InputComponent;
