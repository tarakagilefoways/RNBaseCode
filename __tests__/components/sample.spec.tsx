import React from 'react';
import renderer from 'react-test-renderer';
import SampleComponent from '../../src/components/SampleComponent/index';
import { render } from '@testing-library/react-native';

describe('Sample Componenet Test Suit:', () => {
  test('Renders Correctly and Create Snapshot', () => {
    const tree = renderer.create(<SampleComponent value="Test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Get Text Component with `Test` and Check It Exist', () => {
    const component = render(<SampleComponent value="Test" />);
    expect(component.getByDisplayValue('Test')).toBeTruthy();
  });
});
