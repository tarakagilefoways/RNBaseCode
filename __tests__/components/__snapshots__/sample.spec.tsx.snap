// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample Componenet Test Suit: Renders Correctly and Create Snapshot 1`] = `
<View>
  <TextInput
    style={
      {
        "alignSelf": "center",
        "borderWidth": 0.5,
        "fontSize": 18,
        "height": 40,
        "marginTop": 8,
        "padding": 4,
        "width": "80%",
      }
    }
    value="Test"
  />
</View>
`;
